Add options to specify XSD schema path for xml checkers

alexmurray opened the pull request:
Wasn't sure which version number to use for these new options - so picked 32 since current version is 31

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/flycheck/flycheck?pullRequest=1272) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/flycheck/flycheck?pullRequest=1272) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/flycheck/flycheck?pullRequest=1272) it.</sub>

fmdkdd commented on doc/languages.rst:
```diff
@@ -1252,12 +1252,20 @@ to view the docstring of the syntax checker.  Likewise, you may use
 
       Check syntax with `XMLStarlet <http://xmlstar.sourceforge.net>`_.
 
+      .. defcustom:: flycheck-xml-xmlstarlet-xsd-path
```
For the documentation, I would rather merge the two options in one `defcustom` block, since they are the same for different checkers.  Look at how `flycheck-clang-args` and `flycheck-gcc-args` are defined in the doc.

fmdkdd commented on flycheck.el:
We are in `31-snapshot`, so that means 31 has not been released yet.  (In doubt, you can also check the release tags and see that the latest one is 30).

fmdkdd commented on flycheck.el:
Same here, version would be 31.

alexmurray commented:
Addressed review comments in https://github.com/flycheck/flycheck/pull/1272/commits/0ed8dc7b92aeaef16384e8656a1eef394ac719bb and https://github.com/flycheck/flycheck/pull/1272/commits/0f0fae12a76b1b6de73dca3f0e8cccdf168eb420

alexmurray commented:
Added CHANGES.rst entry in https://github.com/flycheck/flycheck/pull/1272/commits/e93b4ad6c02d97d99575d38e9d91397a6c367dee

fmdkdd commented:
Great!  Thanks for the swift changes.  I'm going to need one more thing: could you squash all your commits?  We have disabled squashing for preserving GPG signatures (I know, your commits have none, but it's disabled repo-wise).

alexmurray commented:
Squashed into https://github.com/flycheck/flycheck/pull/1272/commits/312b80b561ae4b7cc7a1a43579e16f7b89e004dc

cpitclaudel commented:
Thanks!

fmdkdd commented:
Great!  @alexmurray thank you for your contribution.  :+1:

alexmurray commented:
No worries - thanks for you guys stepping up to maintain flycheck once @lunaryorn left - it is indispensable.
